suites/message/nested.go

```diff
@@ -334,23 +320,26 @@ type msStage struct {
 
 // Creates a multisig actor with its creator as sole approver.
 func prepareStage(td *drivers.TestDriver, creatorBalance, msBalance abi.TokenAmount) *msStage {
-	_, creatorId := td.NewAccountActor(drivers.SECP, creatorBalance)
+	creatorPK, creatorId := td.NewAccountActor(drivers.SECP, creatorBalance)
 
 	msg := td.MessageProducer.CreateMultisigActor(creatorId, []address.Address{creatorId}, 0, 1, chain.Value(msBalance), chain.Nonce(0))
-	result := td.ApplyMessage(msg)
-	require.Equal(td.T, exitcode.Ok, result.Receipt.ExitCode)
-	var ret init_.ExecReturn
-	err := ret.UnmarshalCBOR(bytes.NewReader(result.Receipt.ReturnValue))
+	var initAct init_.State
+	td.GetActorState(builtin.InitActorAddr, &initAct)
+	addr, err := address.NewIDAddress(uint64(initAct.NextID))
 	require.NoError(td.T, err)
 
+	ret := td.ComputeInitActorExecReturn(creatorPK, 0, 0, addr)
+
+	td.ApplyExpect(msg, chain.MustSerialize(&ret))
```

Comment:
This is convoluted. There should be a method with the same functionality as ApplyMessage that just applies a message and returns the result, without necessarily making assertions about that result. I think I exported ApplyMessage (before gas/state validation was implemented) for this purpose.

Reply:
That method would just make assertions about the gas used and resulting state root of the application then?

Reply:
Yes. The important thing is to provide the return value for use later in the test.